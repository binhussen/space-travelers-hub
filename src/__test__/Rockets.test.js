/** @format */

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import missionReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rockets/rockets';
import Rockets from '../pages/Reckets';

const initialState = {
  rockets: [
    {
      id: 'falcon1',
      name: 'Falcon 1',
      reserved: false,
      description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
      image: 'image--rocket1.jpg',
    },
  ],
};

const rootReducer = combineReducers({
  missions: missionReducer,
  rockets: rocketReducer,
});

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});

describe('Rockets tests', () => {
  test('Is Rocket render', () => {
    const rockets = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(rockets).toMatchSnapshot();
  });
  test('Displays the Rocket Page', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    const rocketName = screen.getByText('Falcon 1');
    expect(rocketName).toBeInTheDocument();
  });
});
