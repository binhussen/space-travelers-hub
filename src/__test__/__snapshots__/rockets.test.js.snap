// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets tests Is Rocket render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex items-center flex-col mt-5"
      >
        <div
          class="flex justify-center items-start py-4 gap-5"
        >
          <img
            alt="rocket"
            class="w-25vw rounded"
            src="image--rocket1.jpg"
          />
          <div>
            <h2
              class="font-medium text-2xl"
            >
              Falcon 1
            </h2>
            <p
              class="mt-2 text-lg max-w-45 "
            >
              The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
            </p>
            <button
              class="mt-6 px-3 py-2 bg-sky-500 text-white bottom-0 rounded-md cursor-pointer"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex items-center flex-col mt-5"
    >
      <div
        class="flex justify-center items-start py-4 gap-5"
      >
        <img
          alt="rocket"
          class="w-25vw rounded"
          src="image--rocket1.jpg"
        />
        <div>
          <h2
            class="font-medium text-2xl"
          >
            Falcon 1
          </h2>
          <p
            class="mt-2 text-lg max-w-45 "
          >
            The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
          </p>
          <button
            class="mt-6 px-3 py-2 bg-sky-500 text-white bottom-0 rounded-md cursor-pointer"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
